pkg/importer/dataStream.go

```diff
@@ -125,11 +138,22 @@ func (d dataStream) Close() error {
 func (d *dataStream) dataStreamSelector() (err error) {
 	var r io.Reader
 	scheme := d.Url.Scheme
+	providerType := strings.ToLower(d.provider.Type)
```

Comment:
FWIW, this could be refactored slightly to eliminate an append call and to always "wrap" the returned err:
```
func (d *dataStream) dataStreamSelector() (err error) {
	var r io.Reader
	scheme := d.Url.Scheme
	providerType := strings.ToLower(d.provider.Type)

	switch providerType {
	case "glance":
		r, err = d.glance()
	}

        if r == nil { // no provider
	    switch scheme {
	    case "s3":
		  r, err = d.s3()
	    case "http", "https":
		  r, err = d.http(nil)
	    case "file":
		  r, err = d.local()
	    default:
		  return errors.Errorf("invalid url scheme: %q", scheme)
	    }
        }
	if err == nil && r != nil {
		d.appendReader(rdrTypM[scheme], r)
	}
	return err
}
```

Reply:
Changed to if-else, for validating the providerType as well.